features/steps/steps.py

```diff
@@ -64,36 +64,116 @@ def given_a_machine(context, series):
         CONTAINER_PREFIX + pr_prefix + vm_prefix + series + date_prefix
     )
 
-    context.instance = context.config.cloud_manager.launch(
-        series=series,
-        instance_name=instance_name,
-        image_name=context.series_image_name[series],
-    )
+    context.instances = {
```

Comment:
Given that you are changing context.instance -> context.instances. I think you need to minimally handle the setup of reuse_container which stores its container detection  on `context.instance`. I bet that needs to now be context.instance['uaclient'] right?
 Here are some other cases that need handling too I think
```bash
$ fgrep -r context\.instance features/ | grep -v instances
features/environment.py:                    result = context.instance.execute(
features/environment.py:                result = context.instance.execute(cmd, use_sudo=True)
features/steps/steps.py:        context.instance = context.config.cloud_api.get_instance(
features/steps/steps.py:            context.instance = context.config.cloud_api.get_instance(
```

Reply:
Great catch, yes those cases should all be handled